### PR TITLE
[css-display-3] Add tests for dynamic changes of display: contents pseudo-element styles.

### PR DESCRIPTION
**Pull request by emilio**
These are just some basic smoketests for it, and the interactions with first-letter that were somewhat problematic in Blink.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/5343)

### CODE REVIEW

**Comment by emilio** [comment]
@rune-opera or @mrego, could any of you take a look?

Thanks!

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/218484454)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 70c03cc993463952c6b20d5890cc0719510991d0
Using browser at version BuildID 20170403100136; SourceStamp 38894655c89e68bcd8f45d31a0d3005f2c2b53db
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>2 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5343/css/css-display-3/display-contents-dynamic-before-after-001.html">/css/css-display-3/display-contents-dynamic-before-after-001.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | FAIL    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5343/css/css-display-3/display-contents-dynamic-before-after-first-letter-001.html">/css/css-display-3/display-contents-dynamic-before-after-first-letter-001.html</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | FAIL    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/218484455)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 70c03cc993463952c6b20d5890cc0719510991d0
Using browser at version 59.0.3053.3 dev
Starting 10 test iterations
No tests run.

File: css/css-display-3/display-contents-dynamic-before-after-001.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+</style>
+<p>You should see the word PASS in green and no red below.</p>
+<div><!--
+ --><div><div class="p"></div></div><!--
+ --><div></div><!--
+ --><div class="a"></div><!--
+ --><div>S<div class="s"></div></div>
+</div>
+<script>
+window.onload = function() {
```
Comment (rune-opera) [inline css/css-display-3/display-contents-dynamic-before-after-001.html]: Do you need to wait for onload?

File: css/css-display-3/display-contents-dynamic-before-after-first-letter-001.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    div::first-letter { color: red; }
+    div::before { display: contents; content: "PASS"; border: 1px solid red; }
+    .active div::before,
+    .active div::first-letter { color: green; }
+</style>
+<p>You should see the word PASS in green and no red below.</p>
+<div></div>
+</div>
+<script>
+window.onload = function() {
```
Comment (rune-opera) [inline css/css-display-3/display-contents-dynamic-before-after-first-letter-001.html]: Do you need to wait for onload?

Comment (emilio) [inline css/css-display-3/display-contents-dynamic-before-after-first-letter-001.html]: I guess not really, ammended both.

**Comment by rune-opera** [comment]
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[css/css-display-3/display-contents-dynamic-before-after-001.html, line 23 at r1](https://reviewable.io:443/reviews/w3c/web-platform-tests/5343#-Kgy8kW84gCWjGECQ3TD-r1-23:-Kh1IbZO7geemn-Z0gpx:b32w0x) ([raw file](https://github.com/w3c/web-platform-tests/blob/ade98089fba7ae6c3dc7db3a2e8555f93c9e6c42/css/css-display-3/display-contents-dynamic-before-after-001.html#L23)):*
<details><summary><i>Previously, rune-opera (Rune Lillesveen) wrote…</i></summary><blockquote>

Do you need to wait for onload?
</blockquote></details>

OK

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/web-platform-tests/5343)*

**Review by ghost — APPROVED** [review]

**Comment by rune-opera** [comment]
:lgtm:

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/web-platform-tests/5343#-:-Kh1PCSCmb72pJAF2G-O:bnfp4nl)*

**Comment by gsnedders** [comment]
Instability is #5406.

**Pull request merged**